src/v/cluster/types.cc

```diff
@@ -37,9 +40,9 @@ bool topic_properties::is_compacted() const {
 
 bool topic_properties::has_overrides() const {
     return cleanup_policy_bitflags || compaction_strategy || segment_size
-           || retention_bytes.has_value() || retention_bytes.is_disabled()
-           || retention_duration.has_value()
-           || retention_duration.is_disabled();
+           || recovery.has_value() || retention_bytes.has_value()
```

Comment:
nit: I would expect that fields in this expression go in the same order as in the structure

Reply:
done